azkaban-common/src/main/java/azkaban/event/Event.java

```diff
@@ -53,24 +54,22 @@ public long getTime() {
     return time;
   }
 
-  public Object getData() {
+  public EventData getData() {
     return eventData;
   }
 
-  public static Event create(Object runner, Type type) {
-    return new Event(runner, type, null, true);
-  }
-
-  public static Event create(Object runner, Type type, Object eventData) {
-    return new Event(runner, type, eventData, true);
+  /**
+   * Creates a new event.
+   *
+   * @param runner runner.
+   * @param type type.
+   * @param eventData EventData, null is not allowed.
+   * @return New Event instance.
+   * @throws NullPointerException if EventData is null.
+   */
+  public static Event create(Object runner, Type type, @NotNull EventData eventData) throws NullPointerException {
```

Comment:
My bad. What you have looks good.
I have approved this change. I am not sure how the new code review approval works. Could you try merging yourself?

If it doesn't work, please let me know and I can merge it for you.
Thanks!